### PR TITLE
Using clang-format

### PR DESCRIPTION
**Pull request by ct2034**

### CODE REVIEW

**Comment by ct2034** [comment]
to format all files run `find . -iname *.h -o -iname *.cpp | xargs clang-format -i`. It searches for the `.clang-format` file in any parent folder.

File: prbt_hardware_support/include/prbt_hardware_support/sto_state_machine.h [hunk 1]
```diff
@@ -335,28 +341,28 @@ class StoStateMachine_ : public msm::front::state_machine_def<StoStateMachine_>
...
-  Row< EnableRequestDuringStop   , halt_done    , Enabling                  , recover_start, none     >,
-  Row< EnableRequestDuringStop   , sto_updated  , Stopping                  , none         , sto_false>,
-  Row< EnableRequestDuringStop   , sto_updated  , none                      , none         , sto_true >
-  // +---------------------------+--------------+---------------------------+--------------+----------+
-  > {};
+  struct transition_table
+    : mpl::vector<
+          //  Start                       Event          Target                      Action         Guard
+          // +---------------------------+--------------+---------------------------+--------------+----------+
+          Row<RobotInactive, sto_updated, Enabling, recover_start, sto_true>,
```
Comment (hslusarek) [inline prbt_hardware_support/include/prbt_hardware_support/sto_state_machine.h]: Is it possible to keep the formatting of such structures, for readability reasons?

File: prbt_hardware_support/test/include/prbt_hardware_support/pilz_manipulator_mock.h [hunk 2]
```diff
@@ -33,39 +33,33 @@
...
+  void advertiseServices(ros::NodeHandle nh, std::string hold_service_name, std::string unhold_service_name,
+                         std::string halt_service_name, std::string recover_service_name);
+
+  MOCK_METHOD2(holdCb, bool(std_srvs::Trigger::Request& req, std_srvs::Trigger::Response& resp));
+  MOCK_METHOD2(unholdCb, bool(std_srvs::Trigger::Request& req, std_srvs::Trigger::Response& resp));
+  MOCK_METHOD2(haltCb, bool(std_srvs::Trigger::Request& req, std_srvs::Trigger::Response& resp));
+  MOCK_METHOD2(recoverCb, bool(std_srvs::Trigger::Request& req, std_srvs::Trigger::Response& resp));
+
+private:
+  bool holdCb_internal(std_srvs::Trigger::Request& req, std_srvs::Trigger::Response& resp);
```
Comment (hslusarek) [inline prbt_hardware_support/test/include/prbt_hardware_support/pilz_manipulator_mock.h:58]: Is it possible to keep the formatting of such structures, for readability reasons?

File: prbt_hardware_support/include/prbt_hardware_support/brake_test_utils.h [hunk 3]
```diff
@@ -26,15 +26,14 @@
 
 #include <prbt_hardware_support/brake_test_utils_exception.h>
 
-static const std::string TOPIC_NAME{"/prbt/joint_states"};
-static constexpr double JOINT_STATES_COMPARISON_FREQUENCY{10.0};
-static constexpr double JOINT_STATES_TOPIC_TIMEOUT{1.0};
-static constexpr double DEFAULT_JOINT_STATES_COMPARISON_TOLERANCE{0.001};
-static constexpr double DEFAULT_ROBOT_MOTION_TIMEOUT_S{1.0};
+static const std::string TOPIC_NAME{ "/prbt/joint_states" };
```
Comment (hslusarek) [inline prbt_hardware_support/include/prbt_hardware_support/brake_test_utils.h:29]: I see no reason for these spaces. What do you think? I would recommend removing this formatting rule.

File: .clang-format [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+AllowShortLoopsOnASingleLine: false
+AllowShortFunctionsOnASingleLine: None
+AllowShortLoopsOnASingleLine: false
+AlwaysBreakTemplateDeclarations: true
+AlwaysBreakBeforeMultilineStrings: false
+BreakBeforeBinaryOperators: false
+BreakBeforeTernaryOperators: false
+BreakConstructorInitializersBeforeComma: true
+BinPackParameters: true
+ColumnLimit: 120
```
Comment (hslusarek) [inline .clang-format:18]: Can we use 80 please. 80 allows for better screen-splitting when studying code.

**Review by hslusarek — CHANGES_REQUESTED** [review]
Only minor things.

File: prbt_hardware_support/test/include/prbt_hardware_support/pilz_manipulator_mock.h (hunk 2 above)
Comment (ct2034) [reply prbt_hardware_support/test/include/prbt_hardware_support/pilz_manipulator_mock.h:58]: yes, with the comments ..
```
// clang-format off
...
// clang-format on
```

File: prbt_hardware_support/include/prbt_hardware_support/brake_test_utils.h (hunk 3 above)
Comment (ct2034) [reply prbt_hardware_support/include/prbt_hardware_support/brake_test_utils.h:29]: Maybe they are supposed to increase readability. But you are right - I will try to disable them.

Comment (ct2034) [reply prbt_hardware_support/include/prbt_hardware_support/brake_test_utils.h:29]: i can not find an option that changes this ... http://releases.llvm.org/5.0.0/tools/clang/docs/ClangFormatStyleOptions.html

File: .clang-format (hunk 4 above)
Comment (ct2034) [reply .clang-format:18]: yes. I am all up for that! it may screw up the code even more, though.

**Comment by ct2034** [comment]
cool .. https://travis-ci.org/PilzDE/pilz_robots/jobs/578808580#L459

**Comment by HiddenBug** [comment]
Due to the fact that it is complicated to introduce clang-format at once for all source files, we suggest the following procedure:
1. choose a package  (currently not under heavy development), apply clang-format, open PR requesting to merge the clang-format changes.
2. continue with step 1 until all packages are formatted
3. activate clang-format
@ct2034 What do you think?

**Comment by jschleicher** [comment]
No progress for several months; heavy conflicts at the moment. Maybe start as suggested in https://github.com/PilzDE/pilz_robots/pull/226#issuecomment-555454899 ? But we'd have to plan that into a corresponding pbi anyways? 
Closing here - no chance to merge in the current state.

**Pull request closed without merging**